tests/custom_lookups/tests.py

```diff
@@ -241,10 +241,10 @@ class CustomExactLookup(models.Lookup):
         OldExactLookup = field.get_lookup('exact')
         author = Author.objects.create(name='author', birthdate=None)
         try:
-            type(field).register_lookup(Exactly, 'exact')
+            field.register_lookup(Exactly, 'exact')
             self.assertEqual(Author.objects.get(birthdate__exact=None), author)
         finally:
-            type(field).register_lookup(OldExactLookup, 'exact')
+            field.register_lookup(OldExactLookup, 'exact')
```

Comment:
This would move the test from a `register_lookup`/`register_lookup` pair of operations to `register_lookup`/`_unregister_lookup`. The latter deletes the lookup rather than overwrites it. With the change, all tests pass, but is that okay for what the test is trying to achieve?

Reply:
Hmm it's not clear to me what this test is trying to accomplish, what's the purpose of `CustomExactLookup` in the first since it's `Exactly` that is registered.

Reply:
@sir-sigurd Can you take a look at this test and tell us what you think? It was added in 10bfa876be59feec24bb6a40fa11bece808ee405 PR #9435 ticket [#28896](https://code.djangoproject.com/ticket/28896)

Should `Exactly` be changed to `CustomExactLookup`? Would using `django.tests.utils.register_lookup` be appropriate here?

Reply:
Either lookup works as a regression test. The important thing is that `lookup_name != 'exact'` as the docstring says.

Reply:
I remember looking at this test when merging 233c70f0479beb3bff9027e6cff680882978fd4d.

I just tested this now and if you use `with register_lookup(field, Exactly, lookup_name='exact'):`, then this is the state at the end of the test:
```
>>> Author._meta.get_field('birthdate').get_lookup('exact')
<class 'custom_lookups.tests.Exactly'>
```
With the current code, the output is `<class 'django.db.models.lookups.Exact'>` which looks correct to me.

So I'd leave this as is and remove the unused `CustomExactLookup`.

Reply:
Sounds good. I've done just that in the latest revision.

Reply:
Thanks for the sleuthing folks!